File: Back_maven/src/main/java/test_projet/model/Culture.java
package test_projet.model;

import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name="culture")
public class Culture {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="id",nullable=false)
	private int id;
	
	
	@Column(nullable=false)
	private int quantite;
	@Column(nullable=false)
	private LocalDate datePlantation;
	@Column(nullable=false)
	private LocalDate dateDernierArrosage;
	@Column(nullable=false)
	private Boolean recolte;
	
	
	@ManyToOne
	@JoinColumn(name="jardin", nullable=false)
	private Jardin jardin;
	
	@ManyToOne
	@JoinColumn(name="plante", nullable=false)
	private Plante plante;
	
	public Culture() {}
	
	public Culture(int quantite, LocalDate datePlantation, LocalDate dateDernierArrosage, Boolean recolte,
			Jardin jardin, Plante plante) {
		this.quantite = quantite;
		this.datePlantation = datePlantation;
		this.dateDernierArrosage = dateDernierArrosage;
		this.recolte = recolte;
		this.jardin = jardin;
		this.plante = plante;
	}

	public int getQuantite() {
		return quantite;
	}

	public void setQuantite(int quantite) {
		this.quantite = quantite;
	}

	public LocalDate getDatePlantation() {
		return datePlantation;
	}

	public void setDatePlantation(LocalDate datePlantation) {
		this.datePlantation = datePlantation;
	}

	public LocalDate getDateDernierArrosage() {
		return dateDernierArrosage;
	}

	public void setDateDernierArrosage(LocalDate dateDernierArrosage) {
		this.dateDernierArrosage = dateDernierArrosage;
	}

	public Boolean getRecolte() {
		return recolte;
	}

	public void setRecolte(Boolean recolte) {
		this.recolte = recolte;
	}

	public Jardin getJardin() {
		return jardin;
	}

	public void setJardin(Jardin jardin) {
		this.jardin = jardin;
	}

	public Plante getPlante() {
		return plante;
	}

	public void setPlante(Plante plante) {
		this.plante = plante;
	}

	@Override
	public String toString() {
		return "Culture [quantite=" + quantite + ", datePlantation=" + datePlantation + ", dateDernierArrosage="
				+ dateDernierArrosage + ", recolte=" + recolte + ", jardin=" + jardin.getNumero() + ", plante=" + plante + "]";
	}
	
	
	
}

File: projet-jardin-angular/src/app/cultures/culture.ts
export class Culture {

    private _id: number = 0;
    private _datePlantation: string;
    private _dateDernierArrosage: string;
    private _quantite: number;
    private _recolte: boolean;
    private _idJardin: number;
    private _idPlante: number;
    private _planteType: string;
  
    constructor(
      datePlantation: string,
      dateDernierArrosage: string,
      quantite: number,
      idJardin: number,
      idPlante: number,
      planteType: string,
      recolte: boolean
    ) {
      this._datePlantation = datePlantation;
      this._dateDernierArrosage = dateDernierArrosage;
      this._quantite = quantite;
      this._idJardin = idJardin;
      this._idPlante = idPlante;
      this._planteType = planteType;
      this._recolte = recolte;
    }
  
    public get id(): number {
      return this._id;
    }
  
    public set id(value: number) {
      this._id = value;
    }
  
    public get datePlantation(): string {
      return this._datePlantation;
    }
  
    public set datePlantation(value: string) {
      this._datePlantation = value;
    }
  
    public get dateDernierArrosage(): string {
      return this._dateDernierArrosage;
    }
  
    public set dateDernierArrosage(value: string) {
      this._dateDernierArrosage = value;
    }
  
    public get quantite(): number {
      return this._quantite;
    }
  
    public set quantite(value: number) {
      this._quantite = value;
    }
  
    public get recolte(): boolean {
      return this._recolte;
    }
  
    public set recolte(value: boolean) {
      this._recolte = value;
    }
  
    public get idJardin(): number {
      return this._idJardin;
    }
  
    public set idJardin(value: number) {
      this._idJardin = value;
    }
  
    public get idPlante(): number {
      return this._idPlante;
    }
  
    public set idPlante(value: number) {
      this._idPlante = value;
    }
  
    public get planteType(): string {
      return this._planteType;
    }
  
    public set planteType(value: string) {
      this._planteType = value;
    }
  }
  